features/323-login-id/api.md

```diff
@@ -0,0 +1,170 @@
+# API Summary
+
+## SDK API
+Following snippets use TypeScript:
+```diff
+- async signup(loginIDs: { [key: string]: string }, password: string, data?: object): Promise<User>;
+- async signupWithUsername(username: string, password: string, data?: object): Promise<User>;
+- async signupWithEmail(email: string, password: string, data?: object): Promise<User>;
++ async signup(
++   loginIDs: { [key: string]: string }[] | { [key: string]: string },
++   password: string, data?: object,
++   realm?: string
++ ): Promise<User>;
++ async signupWithUsername(username: string, password: string, data?: object, realm?: string): Promise<User>;
++ async signupWithEmail(email: string, password: string, data?: object, realm?: string): Promise<User>;
+
+- async login(loginID: { [key: string]: string }, password: string): Promise<User>;
+- async loginWithUsername(username: string, password: string): Promise<User>;
+- async loginWithEmail(email: string, password: string): Promise<User>;
++ async login(loginID: { [key: string]: string } | string, password: string, realm?: string): Promise<User>;
++ async loginWithUsername(username: string, password: string, realm?: string): Promise<User>;
++ async loginWithEmail(email: string, password: string, realm?: string): Promise<User>;
+
++ async requestVerification(loginIDKey: string, loginID: string): Promise<void>;
++ async createLoginID(loginIDKey: string, loginID: string, realm?: string): Promise<void>;
++ async deleteLoginID(loginIDKey: string, loginID: string, realm?: string): Promise<void>;
+
+  async whoami(): Promise<User>;
```

Comment:
As this is PR for #323 , should have email related features's API here too i.e. ForgotPassword, although it shouldn't need much change....

(Besides SDK APIs on client side, if you think there are interfaces on server-side APIs, HTTP response header etc, you can also put it under the "API summary", just FYI)